### PR TITLE
Add spark301db shim layer for WindowInPandas.

### PR DESCRIPTION
**Pull request by firestarman**
Spark300db was removed, so need to add it back to spark301db.

Also adding some metrics.

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
build

**Comment by firestarman** [comment]
If for databricks, do we need to add the shim layer for each spark version ? Such as spark300db, spark301db, or later spark310db ?

**Comment by revans2** [comment]
spark300db is in the process of being removed, but we should look into spark310db just to see if the version from spark301db will work or not.

**Comment by firestarman** [comment]
OK, 
And for this PR, I verified it by running a local Jenkins job over Databricks spark301, and it passed. If no other concerns, it is ready to merge, i think.

**Comment by revans2** [comment]
I think we are missing support for regular 3.1.0 spark and possibly 3.0.1 I am seeing failures in the nightly tests.

File: shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala [hunk 1]
```diff
@@ -133,7 +133,7 @@ class Spark310Shims extends Spark301Shims {
   }
 
   override def getExecs: Map[Class[_ <: SparkPlan], ExecRule[_ <: SparkPlan]] = {
-    Seq(
+    super.getExecs ++ Seq(
```
Comment (firestarman) [inline shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala:136]: Fix the failures of the regular 3.1.0 spark nightly.

**Comment by firestarman** [comment]
rebased

**Comment by firestarman** [comment]
build

File: shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala (hunk 1 above)
Comment (firestarman) [inline shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala:136]: Fix the faliures about window tests for the regular 310 spark.

**Review by firestarman — COMMENTED** [review]
> I think we are missing support for regular 3.1.0 spark and possibly 3.0.1 I am seeing failures in the nightly tests.

Yes.

Spark 300 and 301 nightly are passed. https://blossom.nvidia.com/sw-gpu-spark-jenkins/job/rapids_integration-0.3-github/ 

The new added shim layer in this PR should fix the failures about window tests in databricks 301 spark.

And the change as below should fix the window tests failures for regular 310 spark.

**Review by revans2 — APPROVED** [review]

File: shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala (hunk 1 above)
Comment (revans2) [reply shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala:136]: @tgravescs do you have any issues with this?  If we get breaking changes between 301 and 310 we might need to change this, but I wanted to check with you that you don't see any other issues before merging.

**Review by jlowe — APPROVED** [review]
Merging this to fix the nightly tests.  @tgravescs we can address concerns about pulling in the map of execs from Spark301Shims in a separate PR if that's not desired long-term.

File: shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala (hunk 1 above)
Comment (tgravescs) [reply shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/Spark310Shims.scala:136]: it seems fine until something changes that breaks it.

**Pull request merged**